kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryErrorResponses.java

```diff
@@ -112,7 +113,16 @@ public static boolean isIOError(BigQueryException error) {
     return BigQueryException.UNKNOWN_CODE == error.getCode()
         && error.getCause() instanceof IOException;
   }
-
+  public static boolean isAuthenticationError(BigQueryException error) {
```

Comment:
Could you add java doc for this method please. 
Also could you re-check the conditions - It should be one of the below 2
1. 401 - it is authentication error
2. 400  + (error message matches one of the 5) - it is authentication error.

Reply:
As per [this](https://cloud.google.com/bigquery/docs/error-messages#autherrors),  for the invalid_client type only, HTTP 401 will be returned. The final checks will look like this:

<img width="991" alt="Screenshot 2023-03-02 at 1 55 20 PM" src="https://user-images.githubusercontent.com/121012039/222372637-0a358c96-c2d4-4991-b6bb-19557c07880b.png">

This is pretty granular and I am not sure if this will miss any edge case. Does it look good ?

Reply:
Resolved

Reply:
Wrote unit test for this function